test/framework/controlplane_helpers.go

```diff
@@ -351,3 +357,185 @@ func controlPlaneMachineOptions() []client.ListOption {
 		client.HasLabels{clusterv1.MachineControlPlaneLabelName},
 	}
 }
+
+//UpdateControlplaneNodeDrainTimeoutInput is the input for UpdateControlplaneNodeDrainTimeout
+type UpdateControlplaneNodeDrainTimeoutInput struct {
+	Controlplane               *controlplanev1.KubeadmControlPlane
+	NodeDrainTimeout           *metav1.Duration
+	ClusterProxy               ClusterProxy
+	Cluster                    *clusterv1.Cluster
+	WaitForMachinesToBeUpdated []interface{}
+	ScaleUpTo                  int32
+}
+
+//UpdateControlplaneNodeDrainTimeout updates the NodeDrainTimeout field of controlplane machines
+func UpdateControlplaneNodeDrainTimeout(ctx context.Context, input UpdateControlplaneNodeDrainTimeoutInput) {
+	Expect(ctx).NotTo(BeNil())
+	Expect(input.ClusterProxy).ToNot(BeNil())
+	Expect(input.Cluster).ToNot(BeNil())
+	Expect(input.Controlplane).ToNot(BeNil())
+	controlplane := input.Controlplane
+	mgmtClient := input.ClusterProxy.GetClient()
+
+	patchHelper, err := patch.NewHelper(controlplane, mgmtClient)
+	Expect(err).ToNot(HaveOccurred())
+	controlplane.Spec.NodeDrainTimeout = input.NodeDrainTimeout
+	Expect(patchHelper.Patch(context.TODO(), controlplane)).To(Succeed())
+
+	log.Logf("Manually update NodeDrainTimeout for existing controlplane machines")
+	// for all machine: patch
+	controlplaneMachines := GetControlPlaneMachinesByCluster(ctx, GetControlPlaneMachinesByClusterInput{
+
+		Lister:      mgmtClient,
+		ClusterName: input.Cluster.GetName(),
+		Namespace:   input.Cluster.GetNamespace(),
+	})
+	for i := range controlplaneMachines {
+		patchHelper, err := patch.NewHelper(&controlplaneMachines[i], mgmtClient)
+		Expect(err).ToNot(HaveOccurred())
+		controlplaneMachines[i].Spec.NodeDrainTimeout = input.NodeDrainTimeout
+		Expect(patchHelper.Patch(context.TODO(), &controlplaneMachines[i])).To(Succeed())
+	}
+
+	log.Logf("Add new controlplane machine to see if new machines get new NodeDrainTimeout value")
+	ScaleAndWaitControlPlane(ctx, ScaleAndWaitControlPlaneInput{
```

Comment:
I don't see much benefit of adding the timeout and scaling up controlPlane compared to having node drain value in cluster-template and having 3 control plane machines. These two will go through scale up. 
We can remove `ScaleAndWaitControlPlane` and `UpdateControlplaneNodeDrainTimeout` if we use a custom template flavor.